core/src/main/scala/cats/data/EitherT.scala

```diff
@@ -236,11 +236,78 @@ final case class EitherT[F[_], A, B](value: F[Either[A, B]]) {
 object EitherT extends EitherTInstances with EitherTFunctions
 
 private[data] trait EitherTFunctions {
-  final def left[F[_], A, B](fa: F[A])(implicit F: Functor[F]): EitherT[F, A, B] = EitherT(F.map(fa)(Either.left))
 
-  final def right[F[_], A, B](fb: F[B])(implicit F: Functor[F]): EitherT[F, A, B] = EitherT(F.map(fb)(Either.right))
+  final class LeftPartiallyApplied[B] private[EitherTFunctions] {
```

Comment:
I am curious. Is Scalaz's trick here (add a dummy parameter to this class to make it an `AnyVal`; see [here](https://github.com/scalaz/scalaz/blob/926dd8c380bab865afed8ac1a2d9da2bde488a00/core/src/main/scala/scalaz/EitherT.scala)) necessary to make this zero-cost?